### ISSUE TITLE
About 1 in 10 contract events are not being recognized by my web3 event listeners

### ISSUE THREAD

**Issue by jambtt** [MetaMask/metamask-plugin#1043]
When I interact with my Dapp using Mist, my web3 client receives 100% of events that are broadcast.

When I interact with my Dapp using Metamask (which I love, great work guys!), ~1 in every ~10 is not picked up by my web3 client. I know 100% of the events have fired, because I can view the event in Mist/Wallet.

//solidity event
event LogResult(bytes32 indexed ID, address indexed PlayerAddress, uint indexed Number, uint Result, uint Value, int Status, bytes Proof);

//js
```js
var resultEvent = myC.LogResult({PlayerAddress: web3.eth.defaultAccount}, {
        fromBlock: 'latest'
    });
    resultEvent.watch(function (err, result) {
        if (err) {
            console.log(err)
            return;
        }
        console.log(result);
});
```

**Comment by flyswatter** [comment]
Good catch! May not get to work on until Monday.

I wonder if these are small forks, missing blocks where events were included when the RPC switches to a new fork.

- Dan

**Comment by kumavis** [comment]
I think this is bc our blockTracker tracks the latest block instead of incrementing since the last block it saw. This means we miss blocks when they are shorter than the polling period.

**Comment by kumavis** [comment]
this will be resolved over the next week ( provider-engine refactor )

**Comment by jambtt** [comment]
Absolutely fantastic news. This allows me to make Metamask the recommended way to interact with our dapp.

**Comment by kumavis** [comment]
This was fixed in `12.0.0`

**Comment by kumavis** [comment]
fixed https://github.com/MetaMask/metamask-plugin/blob/master/package.json#L122
